gping/pinger.py

```diff
@@ -102,9 +102,150 @@ def box(self, bottom_left_corner, top_right_corner, paint=None, blank=False):
                 self.line(last_point, point, paint=paint, character=" " if blank else None)
             last_point = point
 
-    @property
-    def lines(self):
-        for line in self.data:
+
+class BasePlot(ContextDecorator):
+    def __init__(self, system, host):
+        self.system = system
+        self.host = host
+        self.buff = deque(maxlen=400)
+
+    def create_canvas(self, width, height):
+        # We space for a newline after each line, so restrict the hight/width a bit
+        width, height = width - 1, height - 1  # Our work area is slightly smaller
+        canvas = Canvas(width, height)
+        # Draw a box around the edges of the screen, one character in.
+        canvas.box(
+            Point(1, 1), Point(width - 1, height - 1)
+        )
+        # We use islice to slice the data because we can't do a ranged slice on a dequeue :(
+        data_slice = list(islice(self.buff, 0, width - 3))
```

Comment:
Is `list()` required here? We only iterate it with `enumerate()`

Reply:
Yes, that is right. I will fix it.